File: processor/src/test/resources/SavedInstanceStateViewModel.java
package com.halcyonmobile.viewmodelfactory.sample;

import android.content.Context;
import androidx.lifecycle.SavedStateHandle;

import com.halcyonmobile.viewmodelfactory.annotation.Provided;
import com.halcyonmobile.viewmodelfactory.annotation.ViewModelFactory;

import javax.inject.Singleton;

import androidx.lifecycle.ViewModel;

/**
 * A simple viewmodel with more than one constructor containing also savedState.
 */
@ViewModelFactory
class SavedInstanceStateViewModel extends ViewModel {
    SavedInstanceStateViewModel(@Provided SavedStateHandle savedStateHandle) {
    }

    SavedInstanceStateViewModel(@Provided Context context, SavedStateHandle savedStateHandle, int x) {
    }

}

File: app/src/main/java/com/halcyonmobile/viewmodelfactory/sample/SavedInstanceStateViewModel.kt
package com.halcyonmobile.viewmodelfactory.sample

import android.content.Context
import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import com.halcyonmobile.viewmodelfactory.annotation.Provided
import com.halcyonmobile.viewmodelfactory.annotation.ViewModelFactory

/**
 * A simple viewmodel with more than one constructor containing also savedState.
 */
@ViewModelFactory
internal class SavedInstanceStateViewModel : ViewModel {

    constructor(savedStateHandle: SavedStateHandle?) {}

    constructor(@Provided context: Context?, savedStateHandle: SavedStateHandle?, x: Int) {}
}